### PR TITLE
Update prewalk and postwalk documentation

### PR DESCRIPTION
**Pull request by iaguirre88**
I realized that the example for `prewalk/2` is a bit confusing since some people might assume that `5 + 3 * 7` is transformed to `5 * 3 + 7` while in reality it's transformed into something like `5 * (3 + 7)` because the tree shape is not changed.

I think by showing the previous and new AST in the example can be more helpful for the developer.

I also added an example for `prewalk/3` and included some references to these functions in the `postwalk/2` and `postwalk/3`

### CODE REVIEW

File: lib/elixir/lib/macro.ex [hunk 1]
```diff
@@ -537,23 +539,43 @@ defmodule Macro do
   @doc """
   Performs a depth-first, pre-order traversal of quoted expressions
   using an accumulator.
+
+  Returns a tuple where the first element is a new ast where each node is the
```
Comment (eksperimental) [inline lib/elixir/lib/macro.ex:543]:
```suggestion
  Returns a tuple where the first element is a new AST where each node is the
```

File: lib/elixir/lib/macro.ex [hunk 2]
```diff
@@ -518,11 +518,13 @@ defmodule Macro do
   ## Examples
 
       iex> ast = quote do: 5 + 3 * 7
+      {:+, _, [5, {:*, _, [3, 7]}]} = ast
```
Comment (josevalim) [inline lib/elixir/lib/macro.ex:521]:
```suggestion
      iex> {:+, _, [5, {:*, _, [3, 7]}]} = ast
```

File: lib/elixir/lib/macro.ex [hunk 3]
```diff
@@ -537,23 +539,43 @@ defmodule Macro do
...
   using an accumulator.
+
+  Returns a tuple where the first element is a new ast where each node is the
+  result of invoking `fun` on each corresponding node and the second one is the
+  final accumulator.
+
+  ## Examples
+
+      iex> ast = quote do: 5 + 3 * 7
+      {:+, _, [5, {:*, _, [3, 7]}]} = ast
```
Comment (josevalim) [inline lib/elixir/lib/macro.ex:550]:
```suggestion
      iex> {:+, _, [5, {:*, _, [3, 7]}]} = ast
```

File: lib/elixir/lib/macro.ex [hunk 4]
```diff
@@ -537,23 +539,43 @@ defmodule Macro do
...
+      iex> {new_ast, acc} = Macro.prewalk(ast, [], fn
+      ...>   {:+, meta, children}, acc -> {{:*, meta, children}, [:+ | acc]}
+      ...>   {:*, meta, children}, acc -> {{:+, meta, children}, [:* | acc]}
+      ...>   other, acc -> {other, acc}
+      ...> end)
+      {{:*, _, [5, {:+, _, [3, 7]}]}, [:*, :+]} = {new_ast, acc}
```
Comment (josevalim) [inline lib/elixir/lib/macro.ex:556]:
```suggestion
      iex> {{:*, _, [5, {:+, _, [3, 7]}]}, [:*, :+]} = {new_ast, acc}
```

File: lib/elixir/lib/macro.ex [hunk 5]
```diff
@@ -537,23 +539,43 @@ defmodule Macro do
...
+      iex> Code.eval_quoted(ast)
+      {26, []}
+      iex> Code.eval_quoted(new_ast)
+      {50, []}
```
Comment (josevalim) [inline lib/elixir/lib/macro.ex:560]:
```suggestion
      {50, []}

```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**